### PR TITLE
feat(nasha.dashboard.partition): new accesses tab

### PR DESCRIPTION
**Pull request by y4nnL**
| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/nasha-refresh-snapshots-tab`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #MANAGER-8952
| License          | BSD 3-Clause

- [ ] ~~Try to keep pull requests small so they can be easily reviewed.~~
- [X] Commits are signed-off
- [X] Only FR translations have been updated
- [X] Branch is up-to-date with target branch
- [X] Lint has passed locally
- [X] Standalone app was ran and tested locally
- [X] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~~Breaking change is mentioned in relevant commits~~

### CODE REVIEW

File: packages/manager/modules/manager-components/src/task-tracker/task-tracker-state.component.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    doneLabel: '<',
+    endpoint: '<',
+    errorLabel: '<',
+    heading: '<',
+    interval: '<',
+    onDone: '<',
+    tasks: '<',
+    todoLabel: '<',
+  },
+  template: `
```
Comment (marie-j) [inline packages/manager/modules/manager-components/src/task-tracker/task-tracker-state.component.js:14]: Why don't we provide directly a component here ? Any benefits of using a compileProvider ?

File: packages/manager/modules/nasha/src/nasha.constants.js [hunk 2]
```diff
@@ -18,9 +18,28 @@ export const NASHA_TASK = {
...
 };
 
+export const NASHA_TASK_TRACKER_HEADER_OPEN = `
+  <div class="mb-4">
+    <oui-message data-type="warning" class="d-block mb-4">
+        <span data-translate="nasha_task_tracker_warning"></span>
+    </oui-message>
+`;
+export const NASHA_TASK_TRACKER_HEADER_PARAM = `
+  <strong
```
Comment (marie-j) [inline packages/manager/modules/nasha/src/nasha.constants.js:28]: What is the need here ? Why store this as a constant ?

**Review by marie-j — CHANGES_REQUESTED** [review]

File: packages/manager/modules/manager-components/src/task-tracker/task-tracker.controller.js [hunk 3]
```diff
@@ -1,23 +1,18 @@
 import { POLL_INTERVAL, TASK_STATUS } from './task-tracker.constants';
 
-let uid = 0;
-
 export default class TaskTrackerController {
   /** @ngInject */
```
Comment (marie-j) [inline packages/manager/modules/manager-components/src/task-tracker/task-tracker.controller.js:4]: should be 
```suggestion
  /* @ngInject */
``` ? 
I'm not sure if both syntax are valid

File: packages/manager/modules/nasha/src/components/partition/access/delete/delete.controller.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+export default class NashaComponentsPartitionAccessDeleteController {
+  /* @ngInject */
+  constructor($http) {
+    this.$http = $http;
+  }
+
+  submit() {
+    this.$http
```
Comment (marie-j) [inline packages/manager/modules/nasha/src/components/partition/access/delete/delete.controller.js:8]:
```suggestion
    return this.$http
```

File: packages/manager/modules/nasha/src/components/task-tracker/task-tracker.utils.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+          taskIds: params.tasks.map(({ taskId }) => taskId).join(','),
+        },
+      };
+    }
+    return false;
+  },
+  resolve: {
+    endpoint: /* @ngInject */ (serviceName) =>
+      `/dedicated/nasha/${serviceName}/task`,
+    operation: /* @ngInject */ (tasks) => tasks[0].operation,
```
Comment (marie-j) [inline packages/manager/modules/nasha/src/components/task-tracker/task-tracker.utils.js:23]: why always the first one ? what if there are two tasks ?

File: packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.constants.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
+export const CUSTOM_SNAPSHOT_NAME_PREFIX = 'snap';
+export const CUSTOM_SNAPSHOT_NAME_SEPARATOR = '-';
+export const STATE_NAME = 'nasha.dashboard.partition.accesses';
```
Comment (marie-j) [inline packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.constants.js:3]: Not too fond of storing  ui router related states in constants like this. what's the need ?

File: packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.controller.js [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+        (result, { label, value }) => ({
+          ...result,
+          [value]: label,
+        }),
+        {},
+      ),
+    };
+  }
+
+  get canCreateAccess() {
```
Comment (marie-j) [inline packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.controller.js:45]: Are getters useful in an angularjs controller ? I'd understand the need in a TS/JS class but here I wonder if it is really needed

File: packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.controller.js [hunk 8]
```diff
@@ -0,0 +1,201 @@
...
+          this.authorizedAccesses = [
+            ...ips
+              .filter((x) => !this.accesses.find(({ ip }) => ip === `${x}/32`))
+              .map((ip) => ({ type: 'IP', ip })),
+            ...blocks
+              .filter((block) => !this.accesses.find(({ ip }) => ip === block))
+              .map((block) => ({ type: 'IP/Block', ip: block })),
+          ].sort(
+            ({ ip: ipBlockA }, { ip: ipBlockB }) =>
+              Number(
```
Comment (marie-j) [inline packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.controller.js:137]: Shouldn't this be extracted to a function ? transformation is the same and maybe it could simplify `showAccessForm` logic  slightly

File: packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.controller.js [hunk 9]
```diff
@@ -0,0 +1,201 @@
...
+
+  createAccess() {
+    const { partitionName } = this.partition;
+    const {
+      ip: { ip },
+      type: { value: type },
+    } = this.model;
+
+    this.isCreatingAccess = true;
+    // this.translate('access_created')
```
Comment (marie-j) [inline packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.controller.js:183]:
```suggestion

```

File: packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.controller.js [hunk 10]
```diff
@@ -0,0 +1,201 @@
...
+
+    this.$http
```
Comment (marie-j) [inline packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.controller.js:185]:
```suggestion
   return  this.$http
```
It is a good practice to return promises

File: packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.controller.js [hunk 11]
```diff
@@ -0,0 +1,201 @@
...
+      )
+      .catch((error) => {
+        this.alertError(error);
+      })
+      .finally(() => {
+        this.isCreatingAccess = false;
+      });
+  }
+
+  translate(key, values) {
```
Comment (marie-j) [inline packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.controller.js:198]: IS this still needed ?

File: packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.routing.js [hunk 12]
```diff
@@ -0,0 +1,25 @@
+import { createTaskTrackerStateOptions } from '../../../components/task-tracker';
+import { STATE_NAME } from './accesses.constants';
```
Comment (marie-j) [inline packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.routing.js:2]: As this is routing and ui router related, imo it is better just to create a 
```
const stateName = xxx
```
in this file if it's to avoid duplication instead of mixing it with plain constants

File: packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.template.html [hunk 13]
```diff
@@ -0,0 +1,111 @@
...
+                        data-type="submit"
+                        data-disabled="!$ctrl.canCreateAccess"
+                    >
+                        <span class="oui-icon oui-icon-check"></span>
+                    </oui-button>
+                    <oui-button
+                        data-on-click="$ctrl.hideAccessForm()"
+                        data-disabled="$ctrl.isCreatingAccess || $ctrl.accesses.length === 1"
+                    >
+                        <span class="oui-icon oui-icon-close"></span>
```
Comment (marie-j) [inline packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.template.html:96]:
```suggestion
                        <span class="oui-icon oui-icon-close" aria-hidden="true"></span>
```
also please add a sr-only description if there is no text input

File: packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.template.html [hunk 14]
```diff
@@ -0,0 +1,111 @@
...
+                    class="d-flex justify-content-end align-items-center"
+                >
+                    <oui-button data-on-click="$ctrl.goToDelete($row.ip)">
+                        <span class="oui-icon oui-icon-bin"></span>
+                    </oui-button>
+                </div>
+            </oui-datagrid-column>
+        </oui-datagrid>
+    </form>
+    <div data-ui-view></div>
```
Comment (marie-j) [inline packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.template.html:110]: It is planned to add things below the datagrid ?

File: packages/manager/modules/nasha/src/dashboard/partition/accesses/delete/delete.routing.js [hunk 15]
```diff
@@ -0,0 +1,17 @@
+export default /* @ngInject */ ($stateProvider) => {
+  $stateProvider.state('nasha.dashboard.partition.accesses.delete', {
+    url: '/:ip/delete',
```
Comment (marie-j) [inline packages/manager/modules/nasha/src/dashboard/partition/accesses/delete/delete.routing.js:3]: Makes no sense to have a `/:ip` state ?

File: packages/manager/modules/nasha/src/dashboard/partition/accesses/delete/index.js [hunk 16]
```diff
@@ -0,0 +1,23 @@
...
+
+import '@uirouter/angularjs';
+import 'oclazyload';
+
+const moduleName = 'ovhManagerNashaDashboardPartitionAccessesDeleteLazyLoading';
+
+angular.module(moduleName, ['ui.router', 'oc.lazyLoad']).config(
+  /* @ngInject */ ($stateProvider) => {
+    $stateProvider.state('nasha.dashboard.partition.accesses.delete.**', {
+      url: '/:customSnapshotName/delete',
```
Comment (marie-j) [inline packages/manager/modules/nasha/src/dashboard/partition/accesses/delete/index.js:11]: param names are different between lazyload and state declaration. We should keep the consistency

**Review by marie-j — CHANGES_REQUESTED** [review]

File: packages/manager/modules/manager-components/src/task-tracker/task-tracker.template.html [hunk 17]
```diff
@@ -5,8 +5,12 @@
                 <oui-modal
                     data-heading="{{:: $ctrl.heading }}"
                     data-primary-label="{{:: ('task_tracker_close' | translate) }}"
-                    data-primary-disabled="!$ctrl.isDone && !$ctrl.hasError"
-                    data-primary-action="$ctrl.onDone({ $tasks: $ctrl.tasks })"
+                    data-primary-disabled="!$ctrl.isClosable"
+                    data-primary-action="$ctrl.onDone({
+                        $tasks: $ctrl.tasks,
+                        $success: $ctrl.isDone,
+                        $error: $ctrl.errorMessage,
```
Comment (mohammed-zahaf) [inline packages/manager/modules/manager-components/src/task-tracker/task-tracker.template.html:12]: why using `$` for each property ?

File: packages/manager/modules/manager-components/src/task-tracker/task-tracker.controller.js (hunk 3 above)
Comment (y4nnL) [reply packages/manager/modules/manager-components/src/task-tracker/task-tracker.controller.js:4]: :+1: 

Looks like it works tho
```javascript
TaskTrackerController.$inject = ["$translate", "$timeout", "$http", "TaskTrackerService"];
function TaskTrackerController($translate, $timeout, $http, TaskTrackerService) {
```

File: packages/manager/modules/nasha/src/components/partition/access/delete/delete.controller.js (hunk 4 above)
Comment (y4nnL) [reply packages/manager/modules/nasha/src/components/partition/access/delete/delete.controller.js:8]: :+1:

File: packages/manager/modules/nasha/src/components/task-tracker/task-tracker.utils.js (hunk 5 above)
Comment (y4nnL) [reply packages/manager/modules/nasha/src/components/task-tracker/task-tracker.utils.js:23]: Because all the tasks have the same operation's value

File: packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.constants.js (hunk 6 above)
Comment (y4nnL) [reply packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.constants.js:3]: :+1: 
This has already been done here : https://github.com/ovh/manager/pull/7495/files#diff-4ba66823bb79f831097d103cc4a5bdd67e8fd97c9aa698c60963c3ae4373effb

File: packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.controller.js (hunk 7 above)
Comment (y4nnL) [reply packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.controller.js:45]: An angularjs controller **is a class**
IMO this is perfectly valid because it avoids using a method, it is more readable, it makes much more sense if the returned value is a primitive (in this case, a boolean). The digest cycle works normally. You can see it as if it were a vuejs computed property

Please take a look at https://blog.thoughtspile.tech/2018/09/20/angularjs-service-property-getter/

Another place where I used it in combination with a setter https://github.com/ovh/manager/pull/6864/files#diff-3f3d97fed0e3c4c727b23f1f6423078c9f45b65cd3ae88c0adf1ae54ad259206R43-R49

File: packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.controller.js (hunk 8 above)
Comment (y4nnL) [reply packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.controller.js:137]: :+1:

File: packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.controller.js (hunk 9 above)
Comment (y4nnL) [reply packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.controller.js:183]: :+1: 
This has already been done here : https://github.com/ovh/manager/pull/7494/files#diff-cf6b3d291063f8971c799cc687f64a5dbfa48562f92706aa3343fbd55d52ed27L183

File: packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.controller.js (hunk 10 above)
Comment (y4nnL) [reply packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.controller.js:185]: :+1:

File: packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.controller.js (hunk 11 above)
Comment (y4nnL) [reply packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.controller.js:198]: It is not :+1: 
This has already been done here : https://github.com/ovh/manager/pull/7495/files#diff-cf6b3d291063f8971c799cc687f64a5dbfa48562f92706aa3343fbd55d52ed27L194-L197

File: packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.template.html (hunk 13 above)
Comment (y4nnL) [reply packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.template.html:96]: :+1:

File: packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.template.html (hunk 14 above)
Comment (y4nnL) [reply packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.template.html:110]: Yes there are substates like the deletion state and the task tracking state

File: packages/manager/modules/nasha/src/dashboard/partition/accesses/delete/delete.routing.js (hunk 15 above)
Comment (y4nnL) [reply packages/manager/modules/nasha/src/dashboard/partition/accesses/delete/delete.routing.js:3]: Can you please explain why ?

It simply mirrors the api :
DELETE /dedicated/nasha/{serviceName}/partition/{partitionName}/access/{ip}
If `/:serviceName` or `/:partitionName` make sense, why this doesnt ?

Also we have to keep track of the ip so the page can be refreshed without loosing the information. I could have done something like `/delete?ip` but what's the difference ?

File: packages/manager/modules/nasha/src/dashboard/partition/accesses/delete/index.js (hunk 16 above)
Comment (y4nnL) [reply packages/manager/modules/nasha/src/dashboard/partition/accesses/delete/index.js:11]: Good catch :+1:

File: packages/manager/modules/manager-components/src/task-tracker/task-tracker.template.html (hunk 17 above)
Comment (y4nnL) [reply packages/manager/modules/manager-components/src/task-tracker/task-tracker.template.html:12]: Because these variables are intended to be used inside the `onDone: '&',` angular expression (a.k.a component binding), what I remember is a convention, right ? Like so
```javascript
data-on-done="$success ? doSomethingWith($tasks) : handle($error)"
```

File: packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.template.html (hunk 14 above)
Comment (y4nnL) [reply packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.template.html:124]: We want the datagrid to remain displayed while substates like the one mentioned above are hit as they are modals
If the ui-view is placed on the datagrid's parent, the view will be emptied

File: packages/manager/modules/nasha/src/dashboard/partition/accesses/delete/delete.routing.js (hunk 15 above)
Comment (y4nnL) [reply packages/manager/modules/nasha/src/dashboard/partition/accesses/delete/delete.routing.js:3]: Actually it is not `/:ip` but `/:ip/delete`

File: packages/manager/modules/nasha/src/components/task-tracker/task-tracker.template.html [hunk 18]
```diff
@@ -0,0 +1,22 @@
+<task-tracker
+    data-endpoint="{{ $ctrl.endpoint }}"
+    data-heading="{{:: ('nasha_components_task_tracker_operation_' + $ctrl.operation) | translate }}"
+    data-on-done="$ctrl.onDone($success, $error)"
```
Comment (mohammed-zahaf) [inline packages/manager/modules/nasha/src/components/task-tracker/task-tracker.template.html:4]: are we still need the `$` char ?

related to:
https://github.com/ovh/manager/pull/7447#discussion_r913465697

File: packages/manager/modules/nasha/src/components/partition/snapshot/delete/delete.controller.js [hunk 19]
```diff
@@ -1,25 +1,21 @@
...
 
-    this.$http
-      .delete(
-        `/dedicated/nasha/${serviceName}/partition/${partitionName}/customSnapshot/${customSnapshotName}`,
-      )
-      .then(({ data: { taskId } }) =>
+    return this.$http
+      .delete(`${this.partitionApiUrl}/customSnapshot/${customSnapshotName}`)
+      .catch((error) => this.close({ error }))
+      .then(({ data: task }) =>
```
Comment (mohammed-zahaf) [inline packages/manager/modules/nasha/src/components/partition/snapshot/delete/delete.controller.js:13]: Doing that, it will probably introduce a mistake in HTTP error case.
For this case, the catch function will be executed and also the then function.

The issue will come into the then function, because we don't have a task in error case.
maybe implement then and catch after.

Let me know.

Comment (y4nnL) [reply packages/manager/modules/nasha/src/components/partition/snapshot/delete/delete.controller.js:13]: Sure !
There are others places where I did the same mistake
This has already been taken care of in this PR : https://github.com/ovh/manager/pull/7494/files#diff-3e4b347490f8c2ca396f30c26909f2342ad5f167da17def31f9d689dbd03f911

File: packages/manager/modules/nasha/src/components/task-tracker/task-tracker.controller.js [hunk 20]
```diff
@@ -0,0 +1,24 @@
+export default class NashaComponentsTaskTracker {
+  /* @ngInject */
+  constructor($translate) {
+    this.$translate = $translate;
+  }
+
+  onDone($success, $error) {
```
Comment (mohammed-zahaf) [inline packages/manager/modules/nasha/src/components/task-tracker/task-tracker.controller.js:7]:
```suggestion
  onDone(success, error) {
```

File: packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.controller.js [hunk 21]
```diff
@@ -0,0 +1,189 @@
...
+    this.accesses = [];
+    this.authorizedAccesses = null;
+    this.model = { ip: null, type: null };
+    this.typeOptions = [];
+  }
+
+  $onInit() {
+    this.typeOptions = this.aclTypeEnum
+      .filter((type) =>
+        type === 'readonly' ? this.partition.protocol === 'NFS' : true,
```
[the remark this answers is not in the document]
Comment (y4nnL) [reply packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.controller.js:25]: I agree
Commit https://github.com/ovh/manager/pull/7495/commits/c01b6cd4af5d1dd0537daa19397202572eb7992f

File: packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.controller.js [hunk 22]
```diff
@@ -0,0 +1,189 @@
...
+        this.accesses = data;
+        if (!this.accesses.length) {
+          return this.showAccessForm().then(() => this.accesses);
+        }
+        return this.accesses;
+      })
+      .then((data) => ({
+        data,
+        meta: { totalCount: data.length },
+      }))
```
[the remark this answers is not in the document]
Comment (y4nnL) [reply packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.controller.js:109]: Actually this is intended. This way if an error occurred a message is displayed with an empty table

File: packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.routing.js [hunk 23]
```diff
@@ -0,0 +1,25 @@
...
+
+export default /* @ngInject */ ($stateProvider) => {
+  $stateProvider.state(STATE_NAME, {
+    url: '/accesses',
+    component: 'nashaDashboardPartitionAccesses',
+    resolve: {
+      breadcrumb: () => null,
+      aclTypeEnum: (schema) =>
```
Comment (mohammed-zahaf) [inline packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.routing.js:10]:
```suggestion
      aclTypeEnum: /* @ngInject */ (schema) =>
```

File: packages/manager/modules/nasha/src/dashboard/partition/snapshots/snapshots.controller.js [hunk 24]
```diff
@@ -90,25 +90,29 @@ export default class NashaDashboardPartitionSnapshotsController {
 
       if (!oldType.enabled && newType.enabled) {
         promises.push(
-          this.$http.post(`${this.partitionApiUrl}/snapshot`, {
-            snapshotType: newType.type,
-          }),
+          this.$http
+            .post(`${this.partitionApiUrl}/snapshot`, {
```
Comment (mohammed-zahaf) [inline packages/manager/modules/nasha/src/dashboard/partition/snapshots/snapshots.controller.js:94]: prefer to have a service for HTTP calls.

File: packages/manager/modules/nasha/src/dashboard/partition/snapshots/snapshots.controller.js [hunk 25]
```diff
@@ -90,25 +90,29 @@ export default class NashaDashboardPartitionSnapshotsController {
...
+            })
+            .then(({ data: task }) => task),
         );
       }
 
       if (oldType.enabled && !newType.enabled) {
         promises.push(
-          this.$http.delete(`${this.partitionApiUrl}/snapshot/${newType.type}`),
+          this.$http
+            .delete(`${this.partitionApiUrl}/snapshot/${newType.type}`)
```
Comment (mohammed-zahaf) [inline packages/manager/modules/nasha/src/dashboard/partition/snapshots/snapshots.controller.js:104]: related to this:
https://github.com/ovh/manager/pull/7447/files#r927410615

File: packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.routing.js (hunk 23 above)
Comment (y4nnL) [reply packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.routing.js:10]: :+1: 
This resolve has been removed : https://github.com/ovh/manager/pull/7494/files#diff-c0f115b196232bd2e7d6bb8a515d38990049029d95bc8261d22a9069a6a7eebfL10-L11
(sorry for that)

**Review by mohammed-zahaf — APPROVED** [review]

File: packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.controller.js (hunk 21 above)
[the remark this answers is not in the document]
Comment (marie-j) [reply packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.controller.js:25]: @y4nnL it is easier for reviewer if suggestions are applied within the PR. Given the number of files and the number of branches related to the topic it is hard to follow which review has been taken into account or not
Also this ensure that when merged into release/main branch code matches the quality standard

File: packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.controller.js [hunk 26]
```diff
@@ -0,0 +1,189 @@
+import { ipBlockToNumber } from '../../../nasha.utils';
+import { TRANSLATE } from './accesses.constants';
```
Comment (ghyenne) [inline packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.controller.js:2]:
```suggestion
import { TRANSLATE_PREFIX } from './accesses.constants';
```

File: packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.constants.js [hunk 27]
```diff
@@ -0,0 +1,11 @@
...
+export const TRANSLATE = 'nasha_dashboard_partition_accesses';
+
+export default {
+  CUSTOM_SNAPSHOT_NAME_PREFIX,
+  CUSTOM_SNAPSHOT_NAME_SEPARATOR,
+  STATE_NAME,
+  TRANSLATE,
```
Comment (ghyenne) [inline packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.constants.js:10]:
```suggestion
  TRANSLATE_PREFIX,
```

File: packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.controller.js [hunk 28]
```diff
@@ -0,0 +1,189 @@
...
+      .catch((error) => {
+        this.alertError(error);
+      })
+      .finally(() => {
+        this.isCreatingAccess = false;
+      });
+  }
+
+  translate(key, values) {
+    return this.$translate.instant(`${TRANSLATE}_${key}`, values);
```
Comment (ghyenne) [inline packages/manager/modules/nasha/src/dashboard/partition/accesses/accesses.controller.js:187]:
```suggestion
    return this.$translate.instant(`${TRANSLATE_PREFIX}_${key}`, values);
```

**Review by ghyenne — APPROVED** [review]
